src/visip/dev/extract_signature.py

```diff
@@ -25,30 +25,34 @@ def unwrap_type(type_hint):
         return None   #raise TypeError("Type annotation is required.")
     #elif type_hint is type(None):
     #    return None
-    elif ti.is_any(type_hint):
+    elif isinstance(type_hint, dtype.DTypeBase):
         return type_hint
-    elif ti.is_base_type(type_hint):
-        return type_hint
-    elif ti.is_dataclass(type_hint):
-        return type_hint
-    elif ti.is_newtype(type_hint):
-        return type_hint
-    else:
-        args = ti.get_args(type_hint)
+    elif isinstance(type_hint, dtype.DTypeGeneric):
+        args = type_hint.get_args()
         if args:
             # Is a generic, process its parameters recursively.
             uargs = tuple(unwrap_type(arg) for arg in args)
-            typing_origin = ti.get_typing_origin(type_hint)
-            return typing_origin[uargs]
+            typing_origin = type(type_hint)
+            return typing_origin(*uargs)
         else:
             raise ExcTypeBase(f"Unknown type annotation: {type_hint}")
 
 
 
 def _parameter_from_inspect(param: inspect.Parameter) -> 'ActionParameter':
-    param_type = unwrap_type(param.annotation)
+    assert param.annotation is not None
+    param_type = unwrap_type(dtype.from_typing(param.annotation))
     return ActionParameter(param.name, param_type, param.default, param.kind)
 
+
+def _check_type_var(parameters):
```

Comment:
What is the result of this function? Possibly should be a method of 'Parameters'.